src/components/itineraries/Generate.tsx

```diff
@@ -109,46 +108,53 @@ const Generate: FC = () => {
     create(listsSettings, userIds, num, selfIncluded)
   }
 
+  const [startAddressCaseId, setStartAddressCaseId] = useState<CaseId | null>(null)
```

Comment:
I personally would prefer `undefined` over `null`. And make the next line more explicit (or is the double negation also to catch an empty string?). 
eg: 
``` 
const [startAddressCaseId, setStartAddressCaseId] = useState<CaseId | undefined>() 
const showStartAddress = startAddressCaseId === undefined
```